templates/package.hbs.json

```diff
@@ -26,5 +26,6 @@
   "devDependencies": {
     "typescript": "~3.0.1"
   },
-  "files": ["dist/*"]
+  "files": ["dist/*"],
+  "sideEffects": false
```

Comment:
I'm optimistic about this being a useful thing, but it *could* cause unintended treeshaking if a Quilt package does something like, `import './polyfills';`.

cc @TheMallen @tzvipm - default to optimization 👍 or default to safety 👎 ?